pkg/chunk/storage/factory.go

```diff
@@ -28,12 +28,23 @@ const (
 	StorageEngineTSDB   = "tsdb"
 )
 
+type indexStoreFactories struct {
+	indexClientFactoryFunc IndexClientFactoryFunc
+	tableClientFactoryFunc TableClientFactoryFunc
+}
```

Comment:
Since these functions don't seem to be used together, they could simply be stored in two different maps.

Reply:
They could be used together and in most cases, they would be. TableClient manages the tables while IndexClient manages the index within those tables. If you are defining a custom IndexStore you most likely would need both of them but it would not be the case always. It depends on the use case. The current requirement for boltdb shipper implementation in loki needs both. Does it make sense?

Reply:
What I meant is that only one of two functions is used each time in `factory.go` (in `NewTableClient` and `NewIndexClient`). But this is just nitpicking, if you prefer this version, let's keep it.